File: app/src/main/java/net/veldor/flibustaloader/http/TorStarter.java
package net.veldor.flibustaloader.http;

import android.util.Log;

import com.msopentech.thali.android.toronionproxy.AndroidOnionProxyManager;

import net.veldor.flibustaloader.App;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

public class TorStarter {
    private final AndroidOnionProxyManager Tor;
    public static final int TOTAL_TRIES_PER_TOR_STARTUP = 3;
    public static final int TOTAL_SECONDS_PER_TOR_STARTUP = (int) TimeUnit.MINUTES.toSeconds(1);

    public TorStarter(){
        if (App.getInstance().mLoadedTor.getValue() == null) {
            Tor = new AndroidOnionProxyManager(App.getInstance(), App.TOR_FILES_LOCATION);
            Log.d("surprise", "StartTorWorker doWork 39: tor manager added");
            App.getInstance().mLoadedTor.postValue(Tor);
        } else {
            Tor = App.getInstance().mLoadedTor.getValue();
        }
    }

    public boolean startTor() {
        if(Tor != null){
            // тут- время, которое отводится на попытку запуска
            // количество попыток запуска
            // пытаемся запустить TOR. Если он запустится- вернёт TRUE, Иначе- false
            try {
                return Tor.startWithRepeat(TOTAL_SECONDS_PER_TOR_STARTUP, TOTAL_TRIES_PER_TOR_STARTUP);
            } catch (InterruptedException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        return false;
    }

    public AndroidOnionProxyManager getTor() {
        return Tor;
    }
}

File: app/src/main/java/net/veldor/flibustaloader/http/TorStarter.kt
package net.veldor.flibustaloader.http

import androidx.lifecycle.MutableLiveData
import com.msopentech.thali.android.toronionproxy.AndroidOnionProxyManager
import net.veldor.flibustaloader.App
import java.io.IOException
import java.util.concurrent.TimeUnit

class TorStarter {
    var tor: AndroidOnionProxyManager? = null
    fun startTor(): Boolean {
        if (tor != null) {
            // тут- время, которое отводится на попытку запуска
            // количество попыток запуска
            // пытаемся запустить TOR. Если он запустится- вернёт TRUE, Иначе- false
            try {
                return tor!!.startWithRepeat(
                    TOTAL_SECONDS_PER_TOR_STARTUP,
                    TOTAL_TRIES_PER_TOR_STARTUP
                )
            } catch (e: InterruptedException) {
                App.instance.torException = e
                e.printStackTrace()
            } catch (e: IOException) {
                App.instance.torException = e
                e.printStackTrace()
            }
        }
        return false
    }

    companion object {
        const val TOR_FILES_LOCATION = "torfiles"
        var torStartTry = 0
        const val TOTAL_TRIES_PER_TOR_STARTUP = 3
        private const val TOR_NOT_LAUNCH = 0
        const val TOR_LAUNCH_IN_PROGRESS = 1
        const val TOR_LAUNCH_SUCCESS = 2
        const val TOR_LAUNCH_FAILED = 3
        // хранилище статуса HTTP запроса
        val liveTorLaunchState = MutableLiveData(TOR_NOT_LAUNCH)
        val TOTAL_SECONDS_PER_TOR_STARTUP = TimeUnit.MINUTES.toSeconds(1)
            .toInt()
    }

    init {
        if (App.instance.mLoadedTor.value == null) {
            tor = AndroidOnionProxyManager(App.instance, TOR_FILES_LOCATION)
            App.instance.mLoadedTor.postValue(tor)
        } else {
            tor = App.instance.mLoadedTor.value
        }
    }
}